File: backend/app/schemas/article.py
"""
Article Pydantic schemas
"""
from typing import Optional
from datetime import datetime, date
from uuid import UUID
from pydantic import BaseModel, Field


class ArticleBase(BaseModel):
    """Base article schema"""
    article_id: str = Field(..., min_length=1, max_length=100)
    # article_pk: removed field
    article_number: str = Field(..., min_length=1, max_length=100)
    article_url: Optional[str] = None
    approval_group: Optional[UUID] = None
    title: Optional[str] = None
    info_category: Optional[UUID] = None  # Changed to UUID type
    keywords: Optional[str] = None
    importance: Optional[bool] = None
    publish_start: Optional[date] = None
    publish_end: Optional[date] = None
    target: Optional[str] = Field(None, max_length=100)
    question: Optional[str] = None
    answer: Optional[str] = None
    additional_comment: Optional[str] = None


class ArticleCreate(ArticleBase):
    """Schema for creating articles"""
    pass


class ArticleUpdate(BaseModel):
    """Schema for updating articles"""
    # article_pk: removed field
    article_number: Optional[str] = Field(None, min_length=1, max_length=100)
    article_url: Optional[str] = None
    approval_group: Optional[UUID] = None
    title: Optional[str] = None
    info_category: Optional[UUID] = None  # Changed to UUID type
    keywords: Optional[str] = None
    importance: Optional[bool] = None
    publish_start: Optional[date] = None
    publish_end: Optional[date] = None
    target: Optional[str] = Field(None, max_length=100)
    question: Optional[str] = None
    answer: Optional[str] = None
    additional_comment: Optional[str] = None


class Article(ArticleBase):
    """Schema for article response"""
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/types/article.ts
/**
 * 記事関連の型定義
 * バックエンドの article.py スキーマに対応
 */

export interface ArticleBase {
  article_id: string;
  article_number: string;
  article_url?: string | null;
  approval_group?: string | null; // UUID string
  title?: string | null;
  info_category?: string | null; // UUID string
  keywords?: string | null;
  importance?: boolean | null;
  publish_start?: string | null; // ISO date string
  publish_end?: string | null; // ISO date string
  target?: string | null;
  question?: string | null;
  answer?: string | null;
  additional_comment?: string | null;
}

export interface ArticleCreate extends ArticleBase {
  // 記事作成用（BaseとCreateは同じ）
}

export interface ArticleUpdate {
  article_number?: string;
  article_url?: string | null;
  approval_group?: string | null;
  title?: string | null;
  info_category?: string | null;
  keywords?: string | null;
  importance?: boolean | null;
  publish_start?: string | null;
  publish_end?: string | null;
  target?: string | null;
  question?: string | null;
  answer?: string | null;
  additional_comment?: string | null;
}

export interface Article extends ArticleBase {
  created_at: string;
  updated_at: string;
}

// UI用の拡張型
export interface ArticleWithDetails extends Article {
  approval_group_name?: string;
  category_name?: string;
  revision_count?: number; // この記事に対する修正案の数
  last_revision_date?: string;
}

// 一覧表示用の軽量な型
export interface ArticleSummary {
  article_id: string;
  article_number: string;
  title?: string | null;
  approval_group?: string | null;
  approval_group_name?: string;
  category_name?: string;
  importance?: boolean | null;
  publish_start?: string | null;
  publish_end?: string | null;
  updated_at: string;
}

// 記事選択用の軽量な型
export interface ArticleOption {
  article_id: string;
  article_number: string;
  title?: string | null;
}